src/main/java/ui/BoardNameDialog.java

```diff
@@ -44,60 +49,71 @@ public BoardNameDialog(Preferences prefs, Stage mainStage) {
         addListener();
         initializeInvalidNames();
     }
-    
+
     private static void initializeInvalidNames() {
         Set<String> invalids = new HashSet<String>();
         invalids.add("none"); // TODO possibly extend the set in the future
         invalidNames = Collections.unmodifiableSet(invalids);
     }
-    
+
     private void initializeDialog(Stage mainStage) {
         initOwner(mainStage);
         initModality(Modality.APPLICATION_MODAL);
         setTitle("Save board as");
     }
-    
+
     private void createButtons() {
         ButtonType submitButtonType = new ButtonType("OK", ButtonBar.ButtonData.OK_DONE);
         getDialogPane().getButtonTypes().addAll(submitButtonType, ButtonType.CANCEL);
         submitButton = (Button) getDialogPane().lookupButton(submitButtonType);
+        submitButton.addEventFilter(ActionEvent.ACTION, event -> {
+            if (isBoardNameDuplicate(nameField.getText()) && !showWarningDuplicateName()) {
+                event.consume();
+            }
+        });
         submitButton.setId("boardsavebutton");
-        
+
         setResultConverter(submit -> {
             if (submit == submitButtonType) {
                 return nameField.getText();
             }
             return null;
         });
-        
+
+    }
+
+    private boolean showWarningDuplicateName() {
```

Comment:
Since we are getting a return value here, `show` might not be concise enough. Maybe we can rename this method into `confirmDuplicateBoardName` or something along the way.